common/dynamicconfig/constants.go

```diff
@@ -1254,6 +1254,24 @@ these log lines can be noisy, we want to be able to turn on and sample selective
 		100,
 		`MatchingMaxTaskQueuesInDeployment represents the maximum number of task-queues that can be registered in a single worker deployment version`,
 	)
+	MatchingPollerScalingBacklogAgeScaleUp = NewTaskQueueDurationSetting(
+		"matching.pollerScalingMinimumBacklog",
+		200*time.Millisecond,
+		`MatchingPollerScalingBacklogAgeScaleUp is the minimum backlog size that must be accumulated before 
+a decision to scale up the number of pollers will be issued`,
+	)
+	MatchingPollerScalingSyncMatchWaitTime = NewTaskQueueDurationSetting(
+		"matching.pollerScalingSyncMatchWaitTime",
+		1*time.Second,
+		`MatchingPollerScalingSyncMatchWaitTime is the duration a sync-matched poller must exceed before
+a decision to scale down the number of pollers will be issued`,
+	)
+	MatchingPollerScalingDecisionsPerSecond = NewTaskQueueFloatSetting(
+		"matching.pollerScalingDecisionsPerSecond",
+		10,
+		`MatchingPollerScalingDecisionsPerSecond is the maximum number of scaling decisions that will be issued per
+second by one partition manager`,
```

Comment:
One more note: ideally this would be proportional to worker count, or even a rate limit per worker count. For my projects I'm going to have a "approximate per-key rate limiter" that accepts arbitrary keys but uses only fixed memory. We could use that with poller identity to enforce a rate limit per worker.

Or we can actually just count recent pollers from poller history and multiply the rate limit by that (and then make it like 1 or 2 per worker)

Reply:
+1 on proportionality to worker count. Especially because root is exclusively making the scale down decisions, even if number of partitions change when workers scale out, the scale down decisions will be rate limited by this value globally. 
We don't want the rate of poller scale down be the same (in terms of pollers/sec) when there is one worker vs when there are 1000 workers, right?

Reply:
mmmmm... I'm not sure about this. A lot of my testing I did with just one worker, and fast (Rust, Go, Java) workers can perform easily a huge amount of tasks per second when they're simple. (Which isn't uncommon). So, being able to burst up pollers pretty fast for a _single_ worker is fairly desirable.

Now, I'd agree maybe it should be _even faster_ for more workers, but, I wouldn't want to only do 1 per sec for 1 worker.

Reply:
See `GetAllPollerInfo` and similar functions. You could add new ones to get just a count

Reply:
Thought a bit more about this - I ran some tests where I simply moved the scale down on wait-time to be before any ratelimiting, which seemingly makes good sense anyway. If all those polls actually all did wait that long, they should probably scale down. It works perfectly fine

Then, for scale up, the current ratelimiting seems to be plenty fast enough. 

I would have to add a new ratelimiter implementation to do the multiplicative scaling and maybe it makes more sense to just wait for what @dnr is likely to add.

Reply:
My trick for multiplicative scaling that I was going to suggest: let the rate limit be 10*1e6 with proportional burst, then each poll, call AllowN(1e6 / numPollers)

Reply:
Ah, ok yeah good trick

Reply:
"by one partition manager" -> "by one physical queue manager"?

Also, I think we should say something like: "The effective rate limit would be this value multiplied by the number of active pollers".